### ISSUE TITLE
Optimize Modal headline flexibility

### ISSUE THREAD

**Issue by stephanschroeter** [porsche-design-system/porsche-design-system#1454]
Scope
---
Improve flexibility of headline placement and special layouts for modal component without losing accessibility compliance and decrease visual appearance.

Notes  
---
- In some use cases (finder) a headline is defined in the content to link it to the description. There must be the possibility to define/link the aria-described by of the modal from the content 
- Will close https://github.com/porscheui/porsche-design-system-contribution/issues/57
- Design in Figma: https://www.figma.com/file/SyHcKZGdlk5wjId08Nz5Zu/Design-System-V.2?node-id=28%3A1381

Screens
---
![image](https://user-images.githubusercontent.com/42934679/125459083-aca7c235-97d4-4b79-944a-c796a96d1d17.png)
![image](https://user-images.githubusercontent.com/42934679/127523615-e4b9fab2-90e9-4ea7-a9af-b01ba66fff9e.png)

Acceptance criteria
---
- [x] Enable aria prop for title of modal
- [x] Render headline conditional depending on its value and position
- [x] Close "x" button is positioned absolute according to the viewport definition
- [x] Enable full-with placement of image

Subtasks
---
- [x] Add background for close "x"
- [x] Reposition close "x"
- [x] adjust Padding of focus to be visible inside the background for close "x"
- [x] Offer a styled child class for implement full-with image inside the safezone
- [x] Add validation (warning) if the heading is empty and aria prop is not set
- [x] update docs
- [x]  remove/change header to div (not needed for accessibility
- [x] add new prop to set aria-label, override heading if prop is set
- [x] change aside to div, since role="dialog" is already set
- [x] add e2e tests (setting and removing heading and aria props)
- [x] adjust position of x to top right corner
- [x] provide heading slot
- [x] add heading slot to docu
- [x] add marginright 32px to < xxl 
- [x] add image example
- [x] aria audit for header and slotted heading
- [x] remove subheadline from full width modal example, add bottom padding and top padding to heading
- [x] notify https://porschedev.slack.com/archives/CT7AVHTTQ/p1642515125008200 about release
